esbuild.config.prod.mjs

```diff
@@ -0,0 +1,83 @@
+import * as esbuild from 'esbuild'
+import cssModulesPlugin from 'esbuild-css-modules-plugin'
+import fs from 'fs'
+
+const buildsConfig = [
+  {
+    format: 'esm',
+    outfile: 'dist/react-tooltip.esm.js',
+    minify: false,
+  },
+  {
+    format: 'cjs',
+    outfile: 'dist/react-tooltip.cjs.js',
+    minify: false,
+  },
+  {
+    format: 'iife',
+    outfile: 'dist/react-tooltip.iife.js',
+    minify: false,
+  },
+  {
+    format: 'esm',
+    outfile: 'dist/react-tooltip.js', // for styles be exported as `react-tooltip.css`
+    minify: false,
+  },
+  {
+    format: 'esm',
+    outfile: 'dist/react-tooltip.esm.min.js',
+    minify: true,
+  },
+  {
+    format: 'cjs',
+    outfile: 'dist/react-tooltip.cjs.min.js',
+    minify: true,
+  },
+  {
+    format: 'iife',
+    outfile: 'dist/react-tooltip.iife.min.js',
+    minify: true,
+  },
+  {
+    format: 'esm',
+    outfile: 'dist/react-tooltip.min.js',
+    minify: true,
+  },
+]
+
+const builds = []
+
+buildsConfig.forEach(({ format, outfile, minify }) => {
+  builds.push(
+    esbuild.build({
+      entryPoints: ['./src/index.tsx'],
+      bundle: true,
+      outfile,
+      format,
+      treeShaking: true,
+      minify,
+      external: ['react', 'react-dom', 'prop-types'],
+      plugins: [
+        cssModulesPlugin({
+          // inject: true,
+          v2: true,
+          v2CssModulesOption: {
+            pattern: `react-tooltip__[local]_[hash]`,
+          },
+        }),
+      ],
+    }),
+  )
+})
+
+await Promise.allSettled(builds)
+
+// Remove all unecessary or duplicated files from `dist` folder after build
+fs.unlink('./dist/react-tooltip.cjs.css', () => null) // generated by cjs build
+fs.unlink('./dist/react-tooltip.esm.css', () => null) // generated by esm build
+fs.unlink('./dist/react-tooltip.iife.css', () => null) // generated by iife build
+fs.unlink('./dist/react-tooltip.cjs.min.css', () => null) // generated by minified cjs build
+fs.unlink('./dist/react-tooltip.esm.min.css', () => null) // generated by minified esm build
+fs.unlink('./dist/react-tooltip.iife.min.css', () => null) // generated by minified iife build
+fs.unlink('./dist/react-tooltip.js', () => null) // generated by css build
+fs.unlink('./dist/react-tooltip.min.js', () => null) // generated by minified css build
```

Comment:
this config file probably can be better, I just tried to make the production build work as rollup one

PS: with rollup, we export `umd`, but with esbuild we are exporting `iife`.

Reply:
There's probably a way to disable generating the CSS for some builds, and have just one. Couldn't find it though, so I'm fine with leaving this as is.